src/app/view/components/dot-device-selector/dot-device-selector.component.ts

```diff
@@ -51,4 +61,10 @@ export class DotDeviceSelectorComponent implements OnInit {
             ...devices
         ];
     }
+
+    private setDropdownWidth(devices: DotDevice[]): void {
+        const maxTextOption = _.maxBy(devices, (deviceOption: DotDevice) => deviceOption.name.length).name;
+        const dropdownSize = this.stringPixels.getTextWidth(maxTextOption) + this.arrowDropdownlComponentSize;
+        this.dropdownWidth = `${dropdownSize}px`;
+    }
```

Comment:
we should have a method into StringPixels like this:

getWidth(texts: string[]): number{
       ...
}

and then here just call:

this.dropdownWidth = `${getWidth(...)}px`;

Reply:
Why don't you format code? do you hate us?